lib/location/scan_wifi.c

```diff
@@ -59,6 +59,44 @@ void scan_wifi_details_get(struct location_data_details *details)
 }
 #endif
 
+#if defined(LOCATION_METHOD_WIFI_NET_IF_UPDOWN)
+static int scan_wifi_interface_shutdown(struct net_if *iface)
+{
+	LOG_DBG("Shutting down Wi-Fi interface");
+	int ret;
+
+	if (!net_if_is_admin_up(iface)) {
+		return 0;
+	}
+
+	ret = net_if_down(iface);
```

Comment:
We've discussed a bit if we should put the interface down only if it was down before it was brought up. However, now that there is Kconfig option, this is slightly less important. I'll still have a chat about this internally.

And shutdown of interface from `scan_wifi_init()` would need to be treated as a special case of doing the shutdown anyway.